frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.tsx

```diff
@@ -141,28 +146,40 @@ export const EmbedFrame = ({
   const hasVisibleParameters = visibleParameters.length > 0;
 
   const hasHeader = Boolean(finalName || dashboardTabs) || downloadsEnabled;
-  const canParameterPanelSticky =
+
+  const allowParameterPanelSticky =
     !!dashboard && isParametersWidgetContainersSticky(visibleParameters.length);
+  const shouldApplyParameterPanelThemeChangeTransition =
+    !isParameterPanelStickyStateChanging && isParameterPanelSticky;
 
   return (
     <Root
       hasScroll={hasFrameScroll}
       isBordered={bordered}
       hasVisibleOverflowWhenPriting={isPublicDashboard}
-      className={cx(EmbedFrameS.EmbedFrame, className, {
-        [EmbedFrameS.NoBackground]: !background,
-      })}
+      className={cx(
+        className,
+        EmbedFrameS.EmbedFrame,
+        TransitionS.transitionThemeChange,
+        {
+          [EmbedFrameS.NoBackground]: !background,
+        },
+      )}
       data-testid="embed-frame"
       data-embed-theme={theme}
     >
       <ContentContainer
         id={DASHBOARD_PDF_EXPORT_ROOT_ID}
-        className={EmbedFrameS.WithThemeBackground}
+        className={cx(
+          EmbedFrameS.WithThemeBackground,
+          TransitionS.transitionThemeChange,
+        )}
       >
         {hasHeader && (
           <Header
             className={cx(
               EmbedFrameS.EmbedFrameHeader,
+              TransitionS.transitionThemeChange,
```

Comment:
Let's apply transitionThemeChange via CSS selectors instead of separate classes to avoid cluttering the markup, as mentioned [in this comment](https://github.com/metabase/metabase/pull/52259/files#r1918947451)

Reply:
Done except 2 places where we have to do it conditionally